src/field/field_factory.cxx

```diff
@@ -50,6 +50,13 @@ FieldFactory::FieldFactory(Mesh* localmesh, Options* opt)
     : fieldmesh(localmesh == nullptr ? bout::globals::mesh : localmesh),
       options(opt == nullptr ? Options::getRoot() : opt) {
 
+  // Set options
+  // Note: don't use 'options' here because 'options' is a 'const Options*'
+  // pointer, so this would fail if the "input" section is not present.
+  Options nonconst_options{opt == nullptr ? Options::root() : *opt};
```

Comment:
I think this should be an `Options&` to avoid making a copy.

It's a pity we have to do this, but I can't see a nicer alternative